### PR TITLE
WIP: support pull consumer

### PR DESCRIPTION
**Pull request by aaron-ai**

### CODE REVIEW

File: apache/rocketmq/v2/service.proto [hunk 1]
```diff
@@ -244,6 +244,63 @@ message ChangeInvisibleDurationResponse {
...
+  int32 batch_size = 4;
+  FilterExpression filter_expression = 5;
+  google.protobuf.Duration long_polling_timeout = 6;
+}
+
+message PullMessageResponse {
+  oneof content {
+    Status status = 1;
+    Message message = 2;
+    int64 next_offset = 3;
```
Comment (xdkxlk) [inline apache/rocketmq/v2/service.proto:260]: Do we need delivery_timestamp like ReceiveMessageResponse

Comment (aaron-ai) [reply apache/rocketmq/v2/service.proto:260]: The `delivery_timestamp` in `ReceiveMessageResponse` aims to perf the latency of gRPC, we could add it in the next pull request.

File: apache/rocketmq/v2/service.proto [hunk 2]
```diff
@@ -244,6 +244,63 @@ message ChangeInvisibleDurationResponse {
...
+  int64 offset = 3;
+}
+
+message UpdateOffsetResponse {
+  Status status = 1;
+}
+
+message GetOffsetRequest {
+  Resource group = 1;
+  MessageQueue message_queue = 2;
```
Comment (xdkxlk) [inline apache/rocketmq/v2/service.proto:276]: GetOffsetRequest is not clear. Should we return the max offset or min offset?

File: apache/rocketmq/v2/service.proto [hunk 3]
```diff
@@ -244,6 +244,63 @@ message ChangeInvisibleDurationResponse {
...
+  MessageQueue message_queue = 1;
+  google.protobuf.Timestamp timestamp = 2;
+}
+
+message QueryOffsetByTimestampResponse {
+  Status status = 1;
+  int64 offset = 2;
+}
+
+message QueryOffsetByGroupRequest {
```
Comment (xdkxlk) [inline apache/rocketmq/v2/service.proto:294]: What's the difference between QueryOffsetByGroupRequest and GetOffsetRequest

File: apache/rocketmq/v2/service.proto [hunk 4]
```diff
@@ -244,6 +244,63 @@ message ChangeInvisibleDurationResponse {
...
+  int64 offset = 3;
+  int32 batch_size = 4;
+  FilterExpression filter_expression = 5;
+  google.protobuf.Duration long_polling_timeout = 6;
+}
+
+message PullMessageResponse {
+  oneof content {
+    Status status = 1;
+    Message message = 2;
```
Comment (zhouxinyu) [inline apache/rocketmq/v2/service.proto:259]: Should be repeated to support batch manner?

File: apache/rocketmq/v2/service.proto (hunk 2 above)
Comment (aaron-ai) [reply apache/rocketmq/v2/service.proto:276]: @xdkxlk Can you describe more clearly why we need to return max/min offset?

File: apache/rocketmq/v2/service.proto (hunk 3 above)
Comment (aaron-ai) [reply apache/rocketmq/v2/service.proto]: @xdkxlk QueryOffsetByGroupRequest/QueryOffsetByGroupResponse has been removed.

**Review by xdkxlk — APPROVED** [review]

**Pull request merged**